processing/src/main/java/org/apache/carbondata/processing/merger/CompactionResultSortProcessor.java

```diff
@@ -149,16 +149,26 @@
 
   private CarbonColumn[] noDicAndComplexColumns;
 
+  private boolean needSegmentFile;
```

Comment:
why this boolean is required? if any specific scenario is failing or being handled, please add a comment

Reply:
done